lib/bridge/MatrixHandler.js

```diff
@@ -458,6 +500,29 @@ MatrixHandler.prototype._onAdminMessage = Promise.coroutine(function*(req, event
 });
 
 /**
+ * Called when the AS receives a new Matrix invite/join/leave event.
+ * @param {Object} event : The Matrix member event.
+ * @param {MatrixUser} inviter : The inviter (sender).
+ * @param {MatrixUser} invitee : The invitee (receiver).
+ * @return {Promise} which is resolved/rejected when the request finishes.
+ */
+MatrixHandler.prototype._onMemberEvent = Promise.coroutine(function*(req, event, inviter, invitee) {
+    if (!this._memberTracker) {
+        let matrixClient = this.ircBridge.getAppServiceBridge().getClientFactory().getClientAs();
+
+        this._memberTracker = new StateLookup({
+            client : matrixClient,
+            eventTypes: ['m.room.member']
+        });
+    }
+    else {
+        this._memberTracker.onEvent(event);
+
+        yield Promise.resolve();
```

Comment:
You don't need this because by default it will resolve the promise when the function ends.